samples/common.hpp

```diff
@@ -58,6 +58,20 @@
     }
 #endif
 
+// HIP Host function to find if the device supports f64
+bool isSupportedDevice()
```

Comment:
Can we rename this to isF64Supported() ?

The current name is not applicable to all kernels, especially as a 'common' function